test/functional/feature_utxo_set_hash.py

```diff
@@ -24,9 +24,15 @@ def set_test_params(self):
     def test_deterministic_hash_results(self):
         self.log.info("Test deterministic UTXO set hash results")
 
-        # These depend on the setup_clean_chain option, the chain loaded from the cache
-        assert_equal(self.nodes[0].gettxoutsetinfo()['hash_serialized_2'], "b32ec1dda5a53cd025b95387aad344a801825fe46a60ff952ce26528f01d3be8")
-        assert_equal(self.nodes[0].gettxoutsetinfo("muhash")['muhash'], "dd5ad2a105c2d29495f577245c357409002329b9f4d6182c0af3dc2f462555c8")
+        node = self.nodes[0]
+        mocktime = node.getblockheader(node.getblockhash(0))['time'] + 1
+        node.setmocktime(mocktime)
+        blocks = node.generatetoaddress(10, node.get_deterministic_priv_key().address)
+
+        assert_equal(node.gettxoutsetinfo()['hash_serialized_2'], "9411e505b12619bc0d464ead3c2a14823f53d288380adf3742c7f86b64e3d6f1")
+        assert_equal(node.gettxoutsetinfo("muhash")['muhash'], "75d363bf768ebe52102613792ceeb345364874b86f4579eb8030ff835799ecc9")
+
+        node.invalidateblock(blocks[0])
```

Comment:
Any reason to test determinism separate from consistency? It should be ok to just move the assert_equal down a bit:

```diff
diff --git a/test/functional/feature_utxo_set_hash.py b/test/functional/feature_utxo_set_hash.py
index 233a0ccec7..d25803d883 100755
--- a/test/functional/feature_utxo_set_hash.py
+++ b/test/functional/feature_utxo_set_hash.py
@@ -25,20 +25,9 @@ class UTXOSetHashTest(BitcoinTestFramework):
         self.log.info("Test deterministic UTXO set hash results")
 
         node = self.nodes[0]
+        wallet = MiniWallet(node)
         mocktime = node.getblockheader(node.getblockhash(0))['time'] + 1
         node.setmocktime(mocktime)
-        blocks = node.generatetoaddress(10, node.get_deterministic_priv_key().address)
-
-        assert_equal(node.gettxoutsetinfo()['hash_serialized_2'], "9411e505b12619bc0d464ead3c2a14823f53d288380adf3742c7f86b64e3d6f1")
-        assert_equal(node.gettxoutsetinfo("muhash")['muhash'], "75d363bf768ebe52102613792ceeb345364874b86f4579eb8030ff835799ecc9")
-
-        node.invalidateblock(blocks[0])
-
-    def test_muhash_implementation(self):
-        self.log.info("Test MuHash implementation consistency")
-
-        node = self.nodes[0]
-        wallet = MiniWallet(node)
 
         # Generate 100 blocks and remove the first since we plan to spend its
         # coinbase
@@ -78,10 +67,11 @@ class UTXOSetHashTest(BitcoinTestFramework):
         node_muhash = node.gettxoutsetinfo("muhash")['muhash']
 
         assert_equal(finalized[::-1].hex(), node_muhash)
+        assert_equal(node.gettxoutsetinfo()['hash_serialized_2'], "5b1b44097406226c0eb8e1362cd17a1f346522cf9390a8175a57a5262cb1963f")
+        assert_equal(node.gettxoutsetinfo("muhash")['muhash'], "4b8803075d7151d06fad3e88b68ba726886794873fbfa841d12aefb2cc2b881b")
 
     def run_test(self):
         self.test_deterministic_hash_results()
-        self.test_muhash_implementation()
 
 
 if __name__ == '__main__':

Reply:
makes sense, done